### ISSUE TITLE
THE ERROR:  KeyError: 'ts_door0'

### ISSUE THREAD

**Issue by shanqiiu** [kleimerTU/HumanCentricLayouts#1]
Hello, I am very lucky to read your paper, recently ran your code, encountered the following error, i completely according to the knowledge of Read.me operation, can you reslove it for me ,please?

(LayoutEnhancer) PS D:\codes\HumanCentricLayouts-main> python scene_synthesis.py  D:\codes\HumanCentricLayouts-main\config\3dfront_bedrooms_baseline.yaml
Traceback (most recent call last):ront_bed_baseline - 1 of 10
  File "D:\codes\HumanCentricLayouts-main\scene_synthesis.py", line 222, in <module>
    main(sys.argv[1:])
  File "D:\codes\HumanCentricLayouts-main\scene_synthesis.py", line 156, in main
    nodes, collision, scene_dict= add_furniture(sequence.clone().to('cpu'), minvalue_dict, maxvalue_dict, dict_int2cat, dict_cat2model, dict_cat2dims, model2minp, model2maxp, res=res, nodes=nodes, scene_dict=scene_dict, invalid_models=invalid_models,order_switch=order_switch,collision_check=collision_check)[:3]
  File "D:\codes\HumanCentricLayouts-main\src\reconstruction.py", line 117, in add_furniture
    model_min_p = model2minp[best_model]
KeyError: 'ts_door0'

**Comment by kleimerTU** [comment]
Thanks for the notice. A few of the models we used are not part of the 3DFRONT dataset but taken from elsewhere, so they were missing in the dictionaries.

I updated the preprocessing script to add those entries to the dictionaries. You'll have to either run the preprocessing script again with the --create_dicts flag, or manually add the entries from the updated script to the files model2bbox.json, model2minp.json and model2maxp.json.

**Comment by WHQCHINA** [comment]
Can this method synthesize scenes with house shapes other than rectangles? , such as uneven room shape scenes.

**Comment by shanqiiu** [comment]
Hello, I ran a new preprocessing script immediately after receiving your reply, and the error I got is still the same as the original，
`    parser.add_argument(
        "--create_dicts",
        action="store_true",
        help="Create the bbox dictionaries even if they exist already (this takes a while)"
    )`
And I want to know if the default value here needs to be changed.

**Comment by kleimerTU** [comment]
Did you also update your code with the new version from Github? It will only work with the updated version.

The --create_dicts flag will recreate the dictionaries during preprocessing. The first time you used the preprocessing script, the dictionaries were created with the additional models missing. If you call the script again without the flag, the script will see that the dictionary files already exist and skip this step. The --create_dicts flag tells the script to recreate the dictionaries even if they exist already, which you need to do to add the missing entries to the dictionaries.

In short, it should work if you update the code and then run
`python preprocess_3dfront.py PATH_3DFRONT_HOUSES PATH_3DFRONT_MODELS --create_dicts`
